qiskit/tools/qobj_to_circuits.py

```diff
@@ -23,8 +39,35 @@ def qobj_to_circuits(qobj):
         for x in qobj.experiments:
             if hasattr(x.header, 'compiled_circuit_qasm'):
                 circuits.append(
-                    qc.QuantumCircuit.from_qasm_str(x.header.compiled_circuit_qasm))
+                    qc.QuantumCircuit.from_qasm_str(
+                        x.header.compiled_circuit_qasm))
+            else:
+                quantum_registers = _get_registers(x.header.qubit_labels,
+                                                   qr.QuantumRegister)
+                classical_registers = _get_registers(x.header.clbit_labels,
```

Comment:
I think you can construct the `quantum_registers` and `classical_registers` much more easily if you just look at the `header.qreg_sizes` and `header.creg_sizes` fields.

For example they look like:
```
qreg_sizes=[['qr', 4]]
creg_sizes=[['cr1', 3], ['cr2', 1]]
```
which means:
```
QuantumRegister(size=4, name='qr')
ClassicalRegister(size=3, name='cr1')
ClassicalRegister(size=1, name='cr2')
```

Reply:
Ah, good call. That works and is definitely faster than the multiple loops